src/System.Collections.Specialized/src/System/Collections/Specialized/OrderedDictionary.cs

```diff
@@ -175,21 +172,27 @@ object ICollection.SyncRoot
         {
             get
             {
-                return ((DictionaryEntry)objectsArray[index]).Value;
+                if (_objectsArray == null)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(index));
+                }
+                return ((DictionaryEntry)_objectsArray[index]).Value;
             }
             set
             {
                 if (_readOnly)
                 {
                     throw new NotSupportedException(SR.OrderedDictionary_ReadOnly);
                 }
-                if (index < 0 || index >= objectsArray.Count)
+                if (_objectsArray == null || index < 0 || index >= _objectsArray.Count)
                 {
                     throw new ArgumentOutOfRangeException(nameof(index));
                 }
-                object key = ((DictionaryEntry)objectsArray[index]).Key;
-                objectsArray[index] = new DictionaryEntry(key, value);
-                objectsTable[key] = value;
+                EnsureObjectsArray();
```

Comment:
Nit, at this point you know _objectsArray is not null.

Reply:
But it's clearer to leave both in there.

Reply:
Leave both in there.